File: 665 - Non-decreasing Array.py
class Solution:
    def checkPossibility(self, nums: List[int]) -> bool:
        decsk = 0
        enter = False
        for i in range(len(nums) - 1):
            if nums[i] > nums[i + 1]:
                if enter:
                    return False
                enter = True
                if i > 0 and nums[i - 1] > nums[i + 1]:
                    nums[i + 1] = nums[i]
                else:
                    nums[i] = nums[i + 1]
        return True

File: 665 - Non-decreasing Array.java
class Solution {
    public boolean checkPossibility(int[] nums) {
        boolean entered = false;
        for (int i = 0; i < nums.length - 1; i++) {
            if (nums[i] > nums[i + 1]) {
                if (entered) {
                    return false;
                }
                entered = true;
                if (i > 0 && nums[i - 1] > nums[i + 1]) {
                    nums[i + 1] = nums[i];
                } else {
                    nums[i] = nums[i + 1];
                }
            }
        }
        return true;
    }
}